### ISSUE TITLE
Nested rules

### ISSUE THREAD

**Issue by steinuil** [queer/boxxy#16]
I'm trying to clean up the mess of configuration directories left by VS Code and I was wondering if it would be possible to use "nested" rewrite rules. I tested with this configuration:

```yaml
rules:
- name: VS Code extension manifest
  mode: file
  target: ~/.vscode/extensions/extensions.json
  rewrite: ~/.config/vscode/extensions.json
- name: VS Code extensions
  mode: directory
  target: ~/.vscode/extensions
  rewrite: ~/.cache/vscode/extensions
```

And it looks like VS Code is using the file in ~/.cache/vscode/extensions/extensions.json despite the other rule being more "specific". It'd be nice if it was bound to ~/.config/vscode/extensions.json instead and that the rest of the files inside of extensions/ were bound to ~/.cache/vscode/extensions.

Would this be possible/not too complicated to implement in boxxy?

Thanks for the project btw, it's working great for me ❤️

**Comment by queer** [comment]
I *believe* this is due to the ordering of the rules in your configuration; the more-broad rule for `~/.vscode/extensions` is applied *after* the `extensions.json` rule, which would shadow the underlying file. This should be fixable by reordering your rules in the short term; this is definitely a UX shortcoming and I'll investigate automatic rule sorting hopefully-soon ^^

**Comment by steinuil** [comment]
I thought I'd tried to reverse the order of the rules already to test but it turns out I didn't! Reversing them causes a panic:

```
 DEBUG boxxy::enclosure     > processing rule 'VS Code extensions'
 INFO  boxxy::enclosure     > applying rule 'VS Code extensions'
 DEBUG boxxy::enclosure::fs > bind mount "/home/steenuil/.cache/vscode/extensions" onto "/tmp/boxxy-containers/odd-forest-2819/home/steenuil/.vscode/extensions" as rw
 DEBUG boxxy::enclosure::fs > bind mount "/home/steenuil/.cache/vscode/extensions" onto "/tmp/boxxy-containers/odd-forest-2819/home/steenuil/.vscode/extensions"
 INFO  boxxy::enclosure     > * ~/.vscode/extensions -> /home/steenuil/.cache/vscode/extensions
 DEBUG boxxy::enclosure     > rewrote base bath "/home/steenuil/.cache/vscode/extensions" => "/tmp/boxxy-containers/odd-forest-2819/home/steenuil/.vscode/extensions"
 DEBUG boxxy::enclosure     > processing rule 'VS Code extensions manifest'
 INFO  boxxy::enclosure     > applying rule 'VS Code extensions manifest'
 DEBUG boxxy::enclosure::fs > bind mount "/home/steenuil/.config/vscode/extensions.json" onto "/tmp/boxxy-containers/odd-forest-2819/home/steenuil/.vscode/extensions/extensions.json" as rw
 DEBUG boxxy::enclosure::fs > bind mount "/home/steenuil/.config/vscode/extensions.json" onto "/tmp/boxxy-containers/odd-forest-2819/home/steenuil/.vscode/extensions/extensions.json"
The application panicked (crashed).
Message:  called `Result::unwrap()` on an `Err` value: 
   0: ENOENT: No such file or directory

Location:
   src/enclosure/fs.rs:65

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

Consider reporting this error using this URL: https://github.com/queer/boxxy/issues/new?title=%3Cautogenerated-issue%3E&body=%23%23+Error%0A%60%60%60%0AError%3A+%0A+++0%3A+ENOENT%3A+No+such+file+or+directory%0A%60%60%60%0A%0A%23%23+Metadata%0A%7Ckey%7Cvalue%7C%0A%7C--%7C--%7C%0A%7C**version**%7C0.3.4%7C%0A
Location: src/enclosure/mod.rs:73
```

I'm on version 0.3.4.

**Comment by queer** [comment]
Fascinating! Thanks for the report, will look into it.

**Comment by queer** [comment]
This should be fixed in 0.3.5!
